main.go

```diff
@@ -62,23 +60,41 @@ var (
 
 // Main is the principal function for the binary, wrapped only by `main` for convenience.
 func Main() error {
-	ccResource := "google.com/cc"
-	ccDevicePaths := []string{"/dev/tpmrm0"}
-	ccMeasurmentPaths := []string{"/sys/kernel/security/tpm0/binary_bios_measurements"}
+	// We create a list of specs, one for each device type.
+	allDeviceSpecs := []*deviceplugin.CcDeviceSpec{
+		{
+			// vTPM for standard Confidential VMs
+			Resource:         "google.com/cc",
+			DevicePaths:      []string{"/dev/tpmrm0"},
+			MeasurementPaths: []string{"/sys/kernel/security/tpm0/binary_bios_measurements"},
+			DeviceLimit:      256, // Allow multiple pods to share the vTPM
+		},
+		{
+			// Intel TDX
+			Resource:    "intel.com/tdx",
+			DevicePaths: []string{"/dev/tdx-guest", "/dev/tdx_guest"}, // Some kernels use different names
+			// TDX does not have a separate measurement file, attestation is done via ioctl.
+			MeasurementPaths: []string{},
+			DeviceLimit:      1, // Only one container can use the TDX device at a time per node
```

Comment:
This limitation is that "only container A can access TDX device at time X, but container B can access the device after A is done" right?